### PR TITLE
use ansible-runner for playbook execution

### PR DESCRIPTION
**Pull request by dangel101**
replacing ansible-runner-service for playbook execution with ansible-runner

### CODE REVIEW

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleClientFactory.java [hunk 1]
```diff
@@ -13,6 +13,8 @@ public AnsibleRunnerHttpClient create(AnsibleCommandConfig command) {
         AnsibleRunnerLogger runnerLogger = ansibleCommandLogFileFactory.create(command);
         AnsibleRunnerHttpClient client = new AnsibleRunnerHttpClient();
         client.setLogger(runnerLogger);
+//        client.setUUID(command.getUuid());
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleClientFactory.java:16]: remove?

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java [hunk 2]
```diff
@@ -212,4 +254,80 @@ public AnsibleCommandConfig logFileSuffix(String logFileSuffix) {
...
+            ansibleCommand.add("-i");
+            ansibleCommand.add(String.valueOf(uuid));
+            return ansibleCommand;
+        }
+    }
+
+    private void createInventoryFile() {
+        Path inventoryFile;
+        if (getInventoryFile() == null) {
+//             If hostnames are empty we just don't pass any inventory file:
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java:288]: when does it happen?

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleConstants.java [hunk 3]
```diff
@@ -14,6 +14,35 @@
  * This file hold constants used for ansible execution.
  */
 public class AnsibleConstants {
+//    public static final Path PROJECT_DIR = Paths.get(
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleConstants.java:17]: please clean up comments here as well

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java [hunk 4]
```diff
@@ -140,77 +141,40 @@ public AnsibleReturnValue runCommand(AnsibleCommandConfig command,
                 async);
     }
 
-    public AnsibleReturnValue runCommand(AnsibleCommandConfig command, int timeout, BiConsumer<String, String> fn, boolean async) {
+    public AnsibleReturnValue runCommand(AnsibleCommandConfig commandConfig, int timeout, BiConsumer<String, String> fn, boolean async) {
         if (timeout <= 0) {
             timeout = EngineLocalConfig.getInstance().getInteger("ANSIBLE_PLAYBOOK_EXEC_DEFAULT_TIMEOUT");
         }
 
+        log.info("****** inside runCommand****");
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java:149]: needed? please clean up debug logs

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java [hunk 5]
```diff
@@ -51,240 +38,277 @@
...
+        if (ansibleProcess.exitValue() != 0) {
+            throw new AnsibleRunnerCallException(
+                    "Failed to execute call to cancel playbook. %1$s",
+                    output.toString());
+        }
+    }
+
+    public void runPlaybook(List<String> command, int timeout) throws Exception {
+        final Object executeLock = new Object();
+        log.error("***inside run playbook***");
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java:244]: here as well. and i guess everywhere else;)

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java [hunk 6]
```diff
@@ -51,240 +38,277 @@
...
 
-        return event;
+    private JsonNode getEvent(String eventPath) {
+        // Fetch the event info:
+        JsonNode currentNode = null;
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            String jsonOutput = Files.readString(Paths.get(eventPath), StandardCharsets.UTF_8);
+            if (!jsonIsValid(jsonOutput)) {
+//                throw new AnsibleRunnerCallException(
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java:301]: so what do we do?

File: backend/manager/modules/bll/src/test/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClientTest.java [hunk 7]
```diff
@@ -52,190 +34,190 @@
...
-
-    private static HttpResponse createHttpResponse(String content, int httpStatus) {
-        HttpResponseFactory factory = new DefaultHttpResponseFactory();
-        HttpResponse response = factory.newHttpResponse(
-            new BasicStatusLine(HttpVersion.HTTP_1_1, httpStatus, null), null
-        );
-        response.setEntity(EntityBuilder.create().setText(content).build());
-        return response;
-    }
+//    private static Stream<Arguments> statusesForPlaybook() {
```
Comment (michalskrivanek) [inline backend/manager/modules/bll/src/test/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClientTest.java:37]: clean up please

**Review by michalskrivanek — CHANGES_REQUESTED** [review]

**Comment by mwperina** [comment]
https://github.com/oVirt/ovirt-engine/pull/214

**Pull request closed without merging**